source/Ship.cpp

```diff
@@ -2077,9 +2087,29 @@ bool Ship::CanLand() const
 
 
 
-bool Ship::CannotAct() const
+bool Ship::CannotAct(ActionType actionType) const
 {
-	return (zoom != 1.f || isDisabled || hyperspaceCount || pilotError || cloak);
+	bool canActCloaked = true;
+	if(cloak)
+		switch(actionType)
+		{
+			case ActionType::AFTERBURNER:
+				canActCloaked = attributes.Get("cloaked afterburner");
+			case ActionType::BOARD:
+				canActCloaked = attributes.Get("cloaked boarding");
+			case ActionType::COMMUNICATION:
+				canActCloaked = attributes.Get("cloaked communication");
+			case ActionType::FIRE:
+				canActCloaked = attributes.Get("cloaked firing");
+			case ActionType::PICKUP:
+				canActCloaked = attributes.Get("cloaked pickup");
+			case ActionType::SCAN:
+				canActCloaked = attributes.Get("cloaked scanning");
+		}
+	bool canSendHail = (actionType != ActionType::COMMUNICATION || crew);
+	return (zoom != 1.f || isDisabled || hyperspaceCount || pilotError || !canSendHail
+		|| ((cloak == 1. && !canActCloaked)
+		|| (cloak != 1. && cloak && !cloakDisruption && !canActCloaked)));
```

Comment:
Can you simplify this a bit? Like split it apart. The first part doesn't care about cloaking stuff, so you can check that first thing for example.

Reply:
like this? idk what you mean exactly